src/peft/tuners/adamix.py

```diff
@@ -0,0 +1,416 @@
+# coding=utf-8
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+from dataclasses import asdict, dataclass, field
+from enum import Enum
+from typing import List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..import_utils import is_bnb_available
+from ..utils import (
+    TRANSFORMERS_MODELS_TO_ADAMIX_TARGET_MODULES_MAPPING,
+    ModulesToSaveWrapper,
+    PeftConfig,
+    PeftType,
+    _freeze_adapter,
+    _get_submodules,
+)
+
+
+if is_bnb_available():
+    pass
+
+
+@dataclass
+class AdaMixConfig(PeftConfig):
+    """
+    This is the configuration class to store the configuration of a [`AdaMixModel`].
+
+    Args:
+        target_modules (`Union[List[str],str]`): The names of the modules to apply AdaMix to.
```

Comment:
Please add a few words about what types of layers are amenable to this method. This makes it easier if users want to employ this method for a model that is not among the listed default models.